File: src/plugin/combat/magic/lunar/teleport/WaterbirthTeleportSpell.java
package plugin.combat.magic.lunar.teleport;

import org.redrune.game.global.WorldTile;
import org.redrune.game.content.plugin.combat.spell.type.TeleportSpellPlugin;

/**
 * @author Tyluur <itstyluur@gmail.com>
 * @since 7/28/2017
 */
public class WaterbirthTeleportSpell extends TeleportSpellPlugin {
	
	@Override
	public int levelRequired() {
		return 72;
	}
	
	@Override
	public int[] runesRequired() {
		return arguments(ASTRAL_RUNE, 2, LAW_RUNE, 1, WATER_RUNE, 1);
	}
	
	@Override
	public WorldTile destination() {
		return new WorldTile(2527, 3739, 0);
	}
	
	@Override
	public int spellId() {
		return 47;
	}
	
	@Override
	public double exp() {
		return 72;
	}
	
	@Override
	public MagicBook book() {
		return MagicBook.LUNAR;
	}
}

File: src/main/java/plugin/interaction/combat/magic/lunar/teleport/WaterbirthTeleportSpell.kt
package plugin.interaction.combat.magic.lunar.teleport

import game.content.plugin.combat.spell.type.TeleportSpellPlugin
import game.global.WorldTile
import utility.constants.MagicConstants
import utility.constants.MagicConstants.MagicBook

/**
 * @author Tyluur <itstyluur></itstyluur>@icloud.com>
 * @since 7/28/2017
 */
class WaterbirthTeleportSpell : TeleportSpellPlugin {
    override fun levelRequired(): Int {
        return 72
    }

    override fun runesRequired(): IntArray {
        return arguments(MagicConstants.ASTRAL_RUNE, 2, MagicConstants.LAW_RUNE, 1, MagicConstants.WATER_RUNE, 1)
    }

    override fun destination(): WorldTile {
        return WorldTile(2527, 3739, 0)
    }

    override fun spellId(): Int {
        return 47
    }

    override fun exp(): Double {
        return 72.0
    }

    override fun book(): MagicBook {
        return MagicBook.LUNAR
    }
}